### ISSUE TITLE
Canvas fingerprint not blocked on https://amiunique.org/fp

### ISSUE THREAD

**Issue by srirambv** [brave/browser-ios#1298]
### Description:
Canvas fingerprint not blocked on https://amiunique.org/fp

### Steps to Reproduce

  1. Clean install 1.5 (17.11.06.14)
  2. Enable Fingerprint protection from settings
  3. Visit https://amiunique.org/fp, Canvas fingerprint is not blocked

**Actual result:**

![image](https://user-images.githubusercontent.com/17010094/32496368-e9ca8706-c3c0-11e7-9da6-77f882f3a379.png)

**Expected result:**
Should block canvas fingerprinting

**Reproduces how often:** [Easily reproduced/Intermittent issue]
Easily reproduced 

**Brave Version:**
v1.4.2(17.09.08.16) 

**Device details:**

iPad (iOS 11.1) 
**Website problems only:**
- did you check with Brave Shields down?

- did you check in Puffin/Dolphin (UIWebView-based browsers)? 

- did you check in Safari/Firefox (WkWebView-based browsers)? 

### Additional Information

cc: @diracdeltas

**Comment by jamesmudgett** [comment]
These aren't yet supported and are currently in review by security team. We've got a PR ready with these but needs to be reviewed and merged. Will push to 1.5.2.

**Comment by diracdeltas** [comment]
@jamesmudgett link to code that should be reviewed?

**Comment by jhreis** [comment]
This should currently be covered completely fine. I believe this is just due to the whole limit test set that we have had with EFF tools.

**Comment by jhreis** [comment]
@diracdeltas I believe @jamesmudgett was referring to: https://github.com/brave/browser-ios/issues/1286

We have this ported, sitting on a PR, but I believe this is no longer wanted, is that correct?

**Comment by jamesmudgett** [comment]
@diracdeltas @jhreis has been ported from #1286 laptop fix to iOS here https://github.com/brave/browser-ios/pull/1289

**Comment by diracdeltas** [comment]
yeah, please un-port the fix for https://github.com/brave/browser-ios/issues/1286. it was reverted.

to do this, you can take the current blockCanvasFingerprinting.js script in browser-laptop master.

**Comment by diracdeltas** [comment]
note that #1286 is completely different from this issue though, AFAIK.

**Comment by diracdeltas** [comment]
btw here is what is expected on iOS on https://amiunique.org/fp

![screen shot 2017-12-06 at 6 34 57 pm](https://user-images.githubusercontent.com/549654/33678642-3b84bc06-dab4-11e7-9d5d-2758c0e4193e.png)

**Comment by jhreis** [comment]
@diracdeltas fix was only on a PR, was never merged though, https://github.com/brave/browser-ios/pull/1289 has been closed and again, _not_ merged.

**Comment by diracdeltas** [comment]
@jhreis so, confusingly enough, https://github.com/brave/browser-ios/pull/1289 *does not* port the reverted browser-laptop fix for #1286, so it *would not* have fixed #1286 even if it was merged. it ports an enhancement to fingerprinting that is totally separate from #1286.  

what i would recommend is, again, just copying-pasting the latest fingerprinting block script from browser laptop to get it up-to-date. and then i think you have a separate, iOS-only issue with fingerprinting protection not working at all.
